modules/azure/core/delegate-service-endpoint-join.tf

```diff
@@ -1,5 +1,5 @@
 resource "azurerm_role_definition" "service_endpoint_join" {
-  name  = "role-${var.environment}-${var.location_short}-${var.name}-serviceEndpointJoin"
+  name  = "${var.azure_role_definition_prefix}-${var.environment}-${var.location_short}-${var.name}-serviceEndpointJoin"
```

Comment:
Smart solution to keep backwards compatibility. Might be better in the long run to use unique suffix? If I remember correctly we already add an option to enable/disable this for keyvaults.

Reply:
@phillebaba We could use it, but since we are using azure_ad_group_prefix the same way as I'm implementing here, it seem like the logical way to continue with it. Suffix is in my opinion used for globally unique resources while prefix is used for tenant wide (groups, role definitions etc).

Reply:
Sounds good to me.